test/unit/check-access-test.js

```diff
@@ -92,25 +87,53 @@ test('PermissionSet checkAccess() test - public access', function (t) {
   ps.checkAccess(resourceUrl, randomUser, acl.READ)
     .then(result => {
       t.ok(result, 'Everyone should have inherited read access to file')
+      // Reset the permission set, test a non-default permission
+      ps = new PermissionSet()
+      let auth2 = new Authorization(resourceUrl, !inherit)
+      auth2.setPublic()
+      auth2.addMode(acl.READ)
+      ps.addAuthorization(auth2)
+      return ps.checkAccess(resourceUrl, randomUser, acl.READ)
+    })
+    .then(result => {
+      t.ok(result, 'Everyone should have non-inherited read access to file')
+      t.end()
     })
     .catch(err => {
       console.log(err)
       t.fail(err)
     })
-  // Reset the permission set, test a non-default permission
-  ps = new PermissionSet()
-  let auth2 = new Authorization(resourceUrl, !inherit)
-  auth2.setPublic()
-  auth2.addMode(acl.READ)
-  ps.addAuthorization(auth2)
-  ps.checkAccess(resourceUrl, randomUser, acl.READ)
-    .then(result => {
-      t.ok(result, 'Everyone should have non-inherited read access to file')
+})
+
+test('PermissionSet checkAccess() with remote Group Listings', t => {
+  let groupAclSource = require('../resources/acl-with-group-ttl')
+  let resourceUrl = 'https://alice.example.com/docs/file2.ttl'
+  let aclUrl = 'https://alice.example.com/docs/file2.ttl.acl'
+
+  let groupListingSource = require('../resources/group-listing-ttl')
+  let listingUri = 'https://alice.example.com/work-groups'
+  let groupUri = listingUri + '#Accounting'
+  let fetchGraph = sinon.stub()
+    .returns(parseGraph(rdf, listingUri, groupListingSource))
+  let options = { fetchGraph }
+
+  let bob = 'https://bob.example.com/profile/card#me'
+  let isContainer = false
+  let ps = new PermissionSet(resourceUrl, aclUrl, isContainer, { rdf })
+
+  parseGraph(rdf, aclUrl, groupAclSource)
+    .then(graph => {
+      ps.initFromGraph(graph)
+      return ps.checkAccess(resourceUrl, bob, acl.WRITE, options)
+    })
+    .then(hasAccess => {
+      // External group listings have now been loaded/resolved
+      t.ok(fetchGraph.calledWith(groupUri, options))
+      t.ok(hasAccess, 'Bob should have access as member of group')
```

Comment:
Alice was also given explicit access to the resource as an individual, in that ACL file.